### ISSUE TITLE
Developer or admin disabled listings needs updating

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-frontend#2261]
Visit the Fire.fm listing page on [mobile](https://addons.mozilla.org/android/addon/firefm/) and [desktop](https://addons.mozilla.org/firefox/addon/firefm/) to see the difference. I used the URL to access the page, I don't think it's possible to navigate to it.

Expected: a message indicating the add-on has been disabled by its developers. It'd be good to not show any add-on metadata (icon, developers), since that has been a problem in the past.

**Comment by jvillalobos** [comment]
Pages disabled by an admin, [like this one](https://addons.mozilla.org/android/addon/microsoft-net-framework-assist/), show a "Not Authorized" error. I think, similarly, we should be showing a notice that indicates the add-on was disabled by Mozilla.

**Comment by kumar303** [comment]
What I see on this URL is a Not Authorized error. I don't see any metadata. In this issue are you requesting that we just add a more accurate message about why it was disabled and make sure to hide the metadata?

<img width="715" alt="screenshot 2017-04-11 12 04 44" src="https://cloud.githubusercontent.com/assets/55398/24921048/2858d9a8-1eaf-11e7-98c7-d14e76c52d47.png">

**Comment by jvillalobos** [comment]
Ah, I suppose both cases are being treated the same: Not Authorized if you're not the dev, normal add-on listing if you're the dev (I'm a developer of the add-on in the first example).

Given that, I'm requesting that:
* The page appears the same regardless you being the developer or not.
* Instead of "Not Authorized" explain the add-on was disabled by the developer or Mozilla, depending on the situation. The Suggested Pages section and below all makes sense.

**Comment by muffinresearch** [comment]
Currently disabled addons look very odd:

<img width="1410" alt="add-ons_for_firefox" src="https://user-images.githubusercontent.com/1514/30448738-178b84b0-9987-11e7-83bb-8a56539745d0.png">

So nominating this for MVP as a bug to be fixed.

**Comment by muffinresearch** [comment]
@kumar303 do we actually have any data returned to special case these pages? If we don't we should at least get back to a generic error page that isn't trying to display any addon data, ahead of a more involved fix.

**Comment by kumar303** [comment]
> do we actually have any data returned to special case these pages?

I don't see anything that we can currently use.

Since this is mostly a cosmetic issue for pages that are unlikely to be seen, I think we should defer it until after the MVP release.

**Comment by kumar303** [comment]
oops, I guess I forgot to file the API bug before: https://github.com/mozilla/addons-server/issues/6757

However, notice that the screenshots in https://github.com/mozilla/addons-frontend/issues/2261#issuecomment-329578443 are no longer valid. In the new frontend you will currently see a generic 404 page:

<img width="894" alt="screenshot 2017-10-23 17 14 38" src="https://user-images.githubusercontent.com/55398/31915850-af4cd9e6-b815-11e7-9bc8-a06d367efb21.png">

**Comment by kumar303** [comment]
We have API support for this now \o/ https://github.com/mozilla/addons-server/issues/6757#issuecomment-339634279

**Comment by ValentinaPC** [comment]
@kumar303 - intended result here is returning a 404 page for Developer or admin disabled add-ons?

**Comment by kumar303** [comment]
It depends on the add-on. When a developer goes to their dashboard and sets the add-on to invisible then this disables the add-on. The 404 page should say *disabled by the developer*. When an admin disables the add-on (I don't know how to do this) then the 404 page should say *disabled by an administrator*.

**Comment by ValentinaPC** [comment]
We have the following scenarios:
**1. Invisible add-on**
- set from -> Dev Hub - Manage Status &Versions page
- example: https://addons-dev.allizom.org/en-US/developers/addon/webextcspwebextcspwebextcspweb/versions 
- how it behaves:
![oldpg1](https://user-images.githubusercontent.com/15685960/32325530-08699838-bfd8-11e7-80c3-56712fe8fb43.png)
![newpg1](https://user-images.githubusercontent.com/15685960/32325537-0f0c6800-bfd8-11e7-86c3-dd7fb51057e1.png)

**2. No version add-on**
- set from Dev Hub - Manage Status &Versions page - by deleting all version of an add-on
- example: https://addons-dev.allizom.org/en-US/developers/addon/webext-cspwebext-cspwebext-csp/versions
- how it behaves: add-on is not longer returned at search

**3. Add-on with only version disabled** 
- set from Dev Hub - Manage Status &Versions page - by disabling the only version of an add-on
- example: https://addons-dev.allizom.org/en-US/developers/addon/webextcspwe/versions
- how it behaves:
![oldpg3](https://user-images.githubusercontent.com/15685960/32325640-772dd16c-bfd8-11e7-83da-e3709aae49b6.png)
![newpg3](https://user-images.githubusercontent.com/15685960/32325641-7b0a57f6-bfd8-11e7-9302-fbe3f9fd664f.png) 
-- server error after page refresh

4. Disabled from Admin page
- set from Admin Tools page of any add-on - from https://addons-internal.dev.mozaws.net/en-US/admin/addon/manage/auto-man/ 
- example: https://addons-dev.allizom.org/en-US/developers/addon/auto-man/edit
- how it behaves:
![oldpg4](https://user-images.githubusercontent.com/15685960/32325708-b43e84de-bfd8-11e7-9fc4-dca28a74a83f.png)
![newpg4](https://user-images.githubusercontent.com/15685960/32325712-b7ed1f8c-bfd8-11e7-96c0-85644d323260.png)

I think that at least for case 1 and 4 we should implement the bahavior from old desktop pages to new desktop pages.
@muffinresearch ^

**Comment by muffinresearch** [comment]
@ValentinaPC I think @kumar303 would be best to answer the questions you're raising here. I would have thought that 1 & 4 were the cases this patch was setting out to resolve.

**Comment by ValentinaPC** [comment]
I don't know either. 
I made a supposition after reading the title, kumar's comment and testing how things work in new pages vs old pages after an add-on is disabled.

**Comment by muffinresearch** [comment]
@ValentinaPC I get the expected UI when logged out.

**Comment by ValentinaPC** [comment]
I tried case 1 and 4 while logged out:
1. Search "WebextCSPWebextCSPWebextCSPWeb" and things are the same as logged in:
![loggedout1](https://user-images.githubusercontent.com/15685960/32332716-de762cdc-bfee-11e7-9239-e5f191574752.gif)

4. Use a direct link for this case - https://addons-dev.allizom.org/en-US/firefox/addon/auto-man/ 
![usecase4](https://user-images.githubusercontent.com/15685960/32332983-747fa906-bfef-11e7-934c-29d2fea0e0ed.gif)

**Comment by kumar303** [comment]
I'm having trouble following this thread and figuring out what is still unexpected. Can you summarize the actual vs. expected that caused you to re-open the issue?

The screencast in https://github.com/mozilla/addons-frontend/issues/2261#issuecomment-341449881 is what I expect to happen. The title says *page not found* and the reason is *The add-on has been disabled by an administrator*. Which part of that is unexpected? Are you suggesting that we need a new title? If so, what should the title be? The page no longer exists.

As an aside, it would be helpful for me if you made the screencasts slower. I find it hard to keep up with the sequence of events.

**Comment by muffinresearch** [comment]
@kumar303 My reasons were for re-opening are based @ValentinaPC's screen cast "case 1" above your last comment. Which I could reproduce.

Here's the STR;

* With an account that has privileges login to -dev
* search for the addon "WebextCSPWebextCSPWebextCSPWeb"
* Click the result that matches
* See the full details page.
* Logout
* Go to https://addons-dev.allizom.org/
* If you search for the addon "WebextCSPWebextCSPWebextCSPWeb" on -dev and then click on the result. You see the full details page.
* If you refresh that details page, only then do you see the expected error page.

So I think the issue being seen is more related to access than a specific problem with that page.

One issue that should still be resolved, is that direct linking https://addons-dev.allizom.org/en-US/firefox/addon/webextcspwebextcspwebextcspweb/ should be a 410 gone so that disabled add-ons are removed from search results pages. Although that can filed as a different issue altogether.

**Comment by kumar303** [comment]
The case for logging in with a privileged account will only affect a small percentage of users -- @ValentinaPC can you re-file it as a separate issue? 

Just to keep things moving in our priority queue, I'm going to close this bug since it will ship out on Thursday's push. In other words, I don't see any reason to revert this patch. If we don't revert it then it will ship and leaving the issue open could cause confusion.

**Comment by ValentinaPC** [comment]
I've logged 2 separate issues for case 1 and 4 mentioned above.
@kumar303 : can you explain in few words, what will be pushed with this issue?
Because - as the title mentions "Developer or admin disabled listings needs updating" - these were not implemented.
